lib/ddtrace/utils.rb

```diff
@@ -1,9 +1,21 @@
+require 'thread'
+
 module Datadog
-  # TODO[manu]: write docs
+  # Utils contains low-level utilities, typically to provide pseudo-random trace IDs.
   module Utils
+    # We use a custom random number generator because we want no interference
+    # with the default one. Using the default prng, we could break code that
+    # would rely on srand/rand sequences.
+    @rnd = Random.new
```

Comment:
Calling the generator does alter it because it pulls out a number from the sequence. Example below:

```
srand 1234
puts [ rand, rand ] 
# 0.1915194503788923
# 0.6221087710398319

srand 1234
puts [ rand, rand ]
# 0.1915194503788923
# 0.6221087710398319

srand 1234
# do some tracing without this patch
puts [ rand, rand ]
# unexpected behavior (code relying on srand being predictive is broken)
```